plugins/hls-explicit-record-fields-plugin/src/Ide/Plugin/ExplicitFields.hs

```diff
@@ -0,0 +1,266 @@
+import qualified Development.IDE.Core.Shake      as Shake
+import           Development.IDE.GHC.Compat      (HasSrcSpan (..),
+                                                  HsConDetails (RecCon),
+                                                  HsRecFields (..), HscEnv (..),
+                                                  LPat, Outputable, SrcSpan,
+                                                  pm_mod_summary, unLoc)
+import           Development.IDE.GHC.Compat.Core (Extension (NamedFieldPuns),
+                                                  GhcPass (..),
+                                                  HsExpr (RecordCon, rcon_flds),
+                                                  LHsExpr, ModSummary (..),
+                                                  Pass (..), Pat (..),
+                                                  extensionFlags, hfbPun,
+                                                  hs_valds, mapLoc)
+import           Development.IDE.GHC.Compat.Util (toList)
+import           Development.IDE.GHC.Util        (printOutputable)
+import           Development.IDE.Graph           (RuleResult)
+import           Development.IDE.Graph.Classes   (Hashable, NFData (rnf))
+import           Development.IDE.Spans.Pragmas   (NextPragmaInfo (..),
+                                                  getNextPragmaInfo,
+                                                  insertNewPragma)
+import           Development.IDE.Types.Logger    (cmapWithPrio)
+import           GHC.Generics                    (Generic)
+import           Ide.PluginUtils                 (getNormalizedFilePath,
+                                                  handleMaybeM, pluginResponse,
+                                                  subRange)
+import           Ide.Types                       (PluginDescriptor (..),
+                                                  PluginId (..),
+                                                  PluginMethodHandler,
+                                                  defaultPluginDescriptor,
+                                                  mkPluginHandler)
+import           Language.LSP.Types              (CodeAction (..),
+                                                  CodeActionKind (CodeActionRefactorRewrite),
+                                                  CodeActionParams (..),
+                                                  Command, List (..),
+                                                  Method (..), SMethod (..),
+                                                  TextEdit (..),
+                                                  WorkspaceEdit (WorkspaceEdit),
+                                                  fromNormalizedUri,
+                                                  normalizedFilePathToUri,
+                                                  type (|?) (InR))
+import qualified Language.LSP.Types.Lens         as L
+
+
+data Log = LogShake Shake.Log
```

Comment:
I immediately wonder: do we not want to log anything ourselves?

Reply:
What would be a useful thing to log in your opinion? I have been logging the list of records collected from a given source file for debugging during development, but later removed once I got the plugin working correctly. Do you think it would be a useful thing to provide still?

Reply:
Well, that's what debug logging is for. I can't remember if we have a level below debug at the minute, I'm not sure we do.

I think it's helpful to imagine some user scenarios and consider where logging might help. Probably we'd want to make it debug logging still since otherwise it'll be very noisy. Some ideas:
- I thought I should be able to use the action here but I can't? Might be helpful to see "Found X record usages. No record usages in range for this request" or something.
- I used the action but it did the wrong thing somehow. This is probably covered by the requests (and hence the edit) being logged already.
- ???

Reply:
I have added the list of collected records with their source spans to the debug log. I think this should cover the potential scenarios of "it didn't do what I expected it to".